memcache/memcache.go

```diff
@@ -258,8 +260,10 @@ func (c *Client) dial(addr net.Addr) (net.Conn, error) {
 		cn  net.Conn
 		err error
 	}
-
-	nc, err := net.DialTimeout(addr.Network(), addr.String(), c.netTimeout())
+	if c.DialTimeout == nil {
+		c.DialTimeout = net.DialTimeout
```

Comment:
I think assigning to the `Client` here is bad form - better to use a local variable, or add a function similar to `netTimeout()`